file.go

```diff
@@ -49,11 +51,12 @@ func newFile(dataSources []dataSource, opts LoadOptions) *File {
 		opts.KeyValueDelimiters = "=:"
 	}
 	return &File{
-		BlockMode:   true,
-		dataSources: dataSources,
-		sections:    make(map[string]*Section),
-		sectionList: make([]string, 0, 10),
-		options:     opts,
+		BlockMode:      true,
+		dataSources:    dataSources,
+		sections:       make(map[string][]*Section),
+		sectionList:    make([]string, 0, 10),
+		sectionNumList: make([]int, 0, 10),
```

Comment:
Shouldn't we only initialize this when `opts.AllowNonUniqueSections` is `true`?

Reply:
I have to think about it and test it. I am not sure if it would break something because the code is now using these lists always. If the option is false there are just some errors thrown if the user tries to load non unique sections, if I remember correctly.

Reply:
i added some checks and only initialize the index list if opts.AllowNonUniqueSections is true